pkg/xds/bootstrap/template_v3.go

```diff
@@ -1,168 +1,303 @@
 package bootstrap
 
-const configTemplateV3 string = `
-node:
-  id: {{.Id}}
-  cluster: {{.Service}}
-  metadata:
-{{if .DataplaneToken }}
-    dataplane.token: "{{.DataplaneToken}}"
-{{end}}
-{{if .DataplaneResource}}
-    dataplane.resource: '{{.DataplaneResource}}'
-{{end}}
-{{if .AdminPort }}
-    dataplane.admin.port: "{{ .AdminPort }}"
-{{ end }}
-{{if .DNSPort }}
-    dataplane.dns.port: "{{ .DNSPort }}"
-{{ end }}
-{{if .EmptyDNSPort }}
-    dataplane.dns.empty.port: "{{ .EmptyDNSPort }}"
-{{ end }}
-{{if .ProxyType }}
-    dataplane.proxyType: "{{ .ProxyType }}"
-{{ end }}
-    version:
-      kumaDp:
-        version: "{{ .KumaDpVersion }}"
-        gitTag: "{{ .KumaDpGitTag }}"
-        gitCommit: "{{ .KumaDpGitCommit }}"
-        buildDate: "{{ .KumaDpBuildDate }}"
-      envoy:
-        version: "{{ .EnvoyVersion }}"
-        build: "{{ .EnvoyBuild }}"
-{{if .DynamicMetadata }}
-    dynamicMetadata:
-{{ range $key, $value := .DynamicMetadata }}
-      {{ $key }}: "{{ $value }}"
-{{ end }}
-{{ end }}
+import (
+	"strconv"
 
-{{if .AdminPort }}
-admin:
-  access_log_path: {{ .AdminAccessLogPath }}
-  address:
-    socket_address:
-      protocol: TCP
-      address: "{{ .AdminAddress }}"
-      port_value: {{ .AdminPort }}
-{{ end }}
+	"github.com/asaskevich/govalidator"
+	envoy_bootstrap_v3 "github.com/envoyproxy/go-control-plane/envoy/config/bootstrap/v3"
+	envoy_cluster_v3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoy_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_config_endpoint_v3 "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	envoy_metrics_v3 "github.com/envoyproxy/go-control-plane/envoy/config/metrics/v3"
+	envoy_tls "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3"
+	envoy_type_matcher_v3 "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
```

Comment:
nit: if you use the helpers in `pkg/util/proto/types.go`, you get to simplify imports